### ISSUE TITLE
Compilation error: no member named 'dz' in 'DualStorage

### ISSUE THREAD

**Issue by zwillikon** [AcademySoftwareFoundation/OpenShadingLanguage#1863]
### Problem

I'm currently updating / rebuilding all my Academy Software libraries.

**Expected behavior:**  Successful build

**Actual behavior:**  
```
[33/135] Generating llvm_ops_host_rs.bc
FAILED: src/liboslexec/llvm_ops_host_rs.bc ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host_rs.bc 
cd ./OpenShadingLanguage/src/liboslexec && /usr/bin/clang++ -DOSL_FAST_MATH=1 -DOSL_INTERNAL=1 -DOIIO_DISABLE_DEPRECATED=900000 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DOIIO_HIDE_FORMAT=1 -DOSL_LLVM_VERSION=180 -DOSL_LLVM_FULL_VERSION="18.1.8" -DOSL_LLVM_OPAQUE_POINTERS -DOSL_LLVM_NEW_PASS_MANAGER -Dosoparse=OSL_v1_14_1_osoparse -Doslparse=OSL_v1_14_1_oslparse -std=c++17 -fno-use-cxa-atexit -DOSL_FAST_MATH=1 -DOSL_INTERNAL=1 -DOIIO_DISABLE_DEPRECATED=900000 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DOIIO_HIDE_FORMAT=1 -DOSL_LLVM_VERSION=180 -DOSL_LLVM_FULL_VERSION="18.1.8" -DOSL_LLVM_OPAQUE_POINTERS -DOSL_LLVM_NEW_PASS_MANAGER -Dosoparse=OSL_v1_14_1_osoparse -Doslparse=OSL_v1_14_1_oslparse -std=c++17 -fno-use-cxa-atexit -DOSL_FAST_MATH=1 -DOSL_INTERNAL=1 -DOIIO_DISABLE_DEPRECATED=900000 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DOIIO_HIDE_FORMAT=1 -DOSL_LLVM_VERSION=180 -DOSL_LLVM_FULL_VERSION="18.1.8" -DOSL_LLVM_OPAQUE_POINTERS -DOSL_LLVM_NEW_PASS_MANAGER -Dosoparse=OSL_v1_14_1_osoparse -Doslparse=OSL_v1_14_1_oslparse -std=c++17 -fno-use-cxa-atexit -I./OpenShadingLanguage/src/liboslexec -I./OpenShadingLanguage/src/include -I./OpenShadingLanguage/build/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -DOSL_COMPILING_TO_BITCODE=1 -Wno-deprecated-register -Wno-ignored-attributes -Wno-unknown-attributes -O3 -fno-math-errno -S -emit-llvm -o ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host_rs.s llvm_ops.cpp && /usr/bin/llvm-as -f -o ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host_rs.bc ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host_rs.s
In file included from llvm_ops.cpp:80:
./OpenShadingLanguage/src/include/OSL/dual.h:207:18: error: no member named 'dz' in 'DualStorage<type-parameter-0-0, 3>'
  207 |     , m_dz(other.dz)
      |            ~~~~~ ^
1 error generated.
[35/135] Generating llvm_ops_host.bc
FAILED: src/liboslexec/llvm_ops_host.bc ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host.bc 
cd ./OpenShadingLanguage/src/liboslexec && /usr/bin/clang++ -DOSL_FAST_MATH=1 -DOSL_INTERNAL=1 -DOIIO_DISABLE_DEPRECATED=900000 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DOIIO_HIDE_FORMAT=1 -DOSL_LLVM_VERSION=180 -DOSL_LLVM_FULL_VERSION="18.1.8" -DOSL_LLVM_OPAQUE_POINTERS -DOSL_LLVM_NEW_PASS_MANAGER -Dosoparse=OSL_v1_14_1_osoparse -Doslparse=OSL_v1_14_1_oslparse -std=c++17 -fno-use-cxa-atexit -I./OpenShadingLanguage/src/liboslexec -I./OpenShadingLanguage/src/include -I./OpenShadingLanguage/build/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -DOSL_COMPILING_TO_BITCODE=1 -Wno-deprecated-register -Wno-ignored-attributes -Wno-unknown-attributes -O3 -fno-math-errno -S -emit-llvm -o ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host.s llvm_ops.cpp && /usr/bin/llvm-as -f -o ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host.bc ./OpenShadingLanguage/build/src/liboslexec/llvm_ops_host.s
In file included from llvm_ops.cpp:80:
./OpenShadingLanguage/src/include/OSL/dual.h:207:18: error: no member named 'dz' in 'DualStorage<type-parameter-0-0, 3>'
  207 |     , m_dz(other.dz)
      |            ~~~~~ ^
1 error generated.
[36/135] Generating opfmt_host_rs.bc
FAILED: src/liboslexec/opfmt_host_rs.bc ./OpenShadingLanguage/build/src/liboslexec/opfmt_host_rs.bc 
cd ./OpenShadingLanguage/src/liboslexec && /usr/bin/clang++ -DOSL_FAST_MATH=1 -DOSL_INTERNAL=1 -DOIIO_DISABLE_DEPRECATED=900000 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DOIIO_HIDE_FORMAT=1 -DOSL_LLVM_VERSION=180 -DOSL_LLVM_FULL_VERSION="18.1.8" -DOSL_LLVM_OPAQUE_POINTERS -DOSL_LLVM_NEW_PASS_MANAGER -Dosoparse=OSL_v1_14_1_osoparse -Doslparse=OSL_v1_14_1_oslparse -std=c++17 -fno-use-cxa-atexit -DOSL_FAST_MATH=1 -DOSL_INTERNAL=1 -DOIIO_DISABLE_DEPRECATED=900000 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DOIIO_HIDE_FORMAT=1 -DOSL_LLVM_VERSION=180 -DOSL_LLVM_FULL_VERSION="18.1.8" -DOSL_LLVM_OPAQUE_POINTERS -DOSL_LLVM_NEW_PASS_MANAGER -Dosoparse=OSL_v1_14_1_osoparse -Doslparse=OSL_v1_14_1_oslparse -std=c++17 -fno-use-cxa-atexit -I./OpenShadingLanguage/src/liboslexec -I./OpenShadingLanguage/src/include -I./OpenShadingLanguage/build/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -DOSL_COMPILING_TO_BITCODE=1 -Wno-deprecated-register -Wno-ignored-attributes -Wno-unknown-attributes -O3 -fno-math-errno -S -emit-llvm -o ./OpenShadingLanguage/build/src/liboslexec/opfmt_host_rs.s opfmt.cpp && /usr/bin/llvm-as -f -o ./OpenShadingLanguage/build/src/liboslexec/opfmt_host_rs.bc ./OpenShadingLanguage/build/src/liboslexec/opfmt_host_rs.s
In file included from opfmt.cpp:22:
In file included from ./oslexec_pvt.h:35:
./OpenShadingLanguage/src/include/OSL/dual.h:207:18: error: no member named 'dz' in 'DualStorage<type-parameter-0-0, 3>'
  207 |     , m_dz(other.dz)
      |            ~~~~~ ^
1 error generated.
[38/135] Generating opmatrix_host_rs.bc
FAILED: src/liboslexec/opmatrix_host_rs.bc ./OpenShadingLanguage/build/src/liboslexec/opmatrix_host_rs.bc 
cd ./OpenShadingLanguage/src/liboslexec && /usr/bin/clang++ -DOSL_FAST_MATH=1 -DOSL_INTERNAL=1 -DOIIO_DISABLE_DEPRECATED=900000 -D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -DOIIO_HIDE_FORMAT=1 -DOSL_LLVM_VERSION=180 -DOSL_LLVM_FULL_VERSION="18.1.8" -DOSL_LLVM_OPAQUE_POINTERS -DOSL_LLVM_NEW_PASS_MANAGER -Dosoparse=OSL_v1_14_1_osoparse -Doslparse=OSL_v1_14_1_oslparse -std=c++17 -fno-use-cxa-atexit -I./OpenShadingLanguage/src/liboslexec -I./OpenShadingLanguage/src/include -I./OpenShadingLanguage/build/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -DOSL_COMPILING_TO_BITCODE=1 -Wno-deprecated-register -Wno-ignored-attributes -Wno-unknown-attributes -O3 -fno-math-errno -S -emit-llvm -o ./OpenShadingLanguage/build/src/liboslexec/opmatrix_host_rs.s opmatrix.cpp && /usr/bin/llvm-as -f -o ./OpenShadingLanguage/build/src/liboslexec/opmatrix_host_rs.bc ./OpenShadingLanguage/build/src/liboslexec/opmatrix_host_rs.s
In file included from opmatrix.cpp:13:
In file included from ./oslexec_pvt.h:35:
./OpenShadingLanguage/src/include/OSL/dual.h:207:18: error: no member named 'dz' in 'DualStorage<type-parameter-0-0, 3>'
  207 |     , m_dz(other.dz)
      |            ~~~~~ ^
1 error generated.
[43/135] Building CXX object src/liboslcomp/CMakeFiles/oslcomp.dir/osllex.cpp.o
```

### Steps to Reproduce

```
  cmake \
    -B build \
    -GNinja \
    -DCMAKE_INSTALL_PREFIX=/usr \
    -DCMAKE_INSTALL_LIBDIR=lib \
    -DCMAKE_CXX_STANDARD=17 \
    -DSTOP_ON_WARNING=OFF
  ninja -C build
```

### Versions

* OSL branch/version: master
* OS: Linux
* C++ compiler: GCC 14.2.1 
* LLVM version: 18.1.8
* OIIO version: 2.5.14.0, will update to .15

**Comment by steenax86** [comment]
On line 207 in `dual.h`,` other.dz` needs to be `other.m_dz`. This is being fixed via a PR currently.

**Comment by lgritz** [comment]
Fix proposed in #1871
